browser/ui/views/profiles/brave_avatar_toolbar_button.cc

```diff
@@ -64,14 +64,18 @@ void BraveAvatarToolbarButton::SetHighlight(
   std::u16string revised_highlight_text;
   if (browser_->profile()->IsTor()) {
     revised_highlight_text =
-        l10n_util::GetStringUTF16(IDS_TOR_AVATAR_BUTTON_LABEL);
+        l10n_util::GetStringUTF16(IDS_LOCATION_BAR_OPEN_IN_TOR);
+
+    if (GetWindowCount() > 1) {
+      revised_highlight_text =
+          l10n_util::GetStringFUTF16(IDS_TOR_AVATAR_BUTTON_LABEL,
+                                     base::NumberToString16(GetWindowCount()));
+    }
   } else if (browser_->profile()->IsIncognitoProfile()) {
     // We only want the icon and count for Incognito profiles.
     revised_highlight_text = std::u16string();
-    const int incognito_window_count = delegate_->GetWindowCount();
-
-    if (incognito_window_count > 1) {
-      revised_highlight_text = base::NumberToString16(incognito_window_count);
+    if (GetWindowCount() > 1) {
+      revised_highlight_text = base::NumberToString16(GetWindowCount());
```

Comment:
Instead of calling `GetWindowCount()` four times,
caching count at the start of method and using it all places would be better.

Reply:
cannot cache count because I believe it needs profile. also, `GetWindowCount()` always need to be called on specific profiles because it gives an updated value during runtime?